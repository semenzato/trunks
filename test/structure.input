/* This reinforces the case that the parser can cope with comments. */

// Ditto.

typedef UINT32 foo
typedef UINT16 TPM_TAG
typedef UINT16 TPM_COMMAND_CODE

typedef struct tdBAR
{
  foo x;
  foo y;
  foo z;
} BAR;

typedef struct tdFOOBAR
{
  foo n;
  SIZEIS(n)
    BYTE *key
} FOOBAR;

