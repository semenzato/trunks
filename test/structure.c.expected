/* Copyright (c) 2011 The Chromium OS Authors. All rights reserved.
 * Use of this source code is governed by a BSD-style license that can be
 * found in the LICENSE file.
 */

#include <stdint.h>
#include <string.h>

#include "structure.h"
#include "trunks_internal.h"

void TtlMarshal_BAR(BAR* record, uint8_t** cursorp) {
  TtlMarshal_uint32_t(&record->x, cursorp);
  TtlMarshal_uint32_t(&record->y, cursorp);
  TtlMarshal_uint32_t(&record->z, cursorp);
}

void TtlUnmarshal_BAR(BAR* record, uint8_t** cursorp) {
  TtlUnmarshal_uint32_t(&record->x, cursorp);
  TtlUnmarshal_uint32_t(&record->y, cursorp);
  TtlUnmarshal_uint32_t(&record->z, cursorp);
}

uint32_t TtlSizeCalc_BAR(BAR* record) {
  return (4 + 4 + 4);
}

void TtlMarshal_FOOBAR(FOOBAR* record, uint8_t** cursorp) {
  TtlMarshal_uint32_t(&record->n, cursorp);
  TtlMarshal_Blob(record->key, cursorp, record->n);
}

void TtlUnmarshal_FOOBAR(FOOBAR* record, uint8_t** cursorp) {
  TtlUnmarshal_uint32_t(&record->n, cursorp);
  TtlUnmarshal_Blob(&record->key, cursorp, record->n);
}

uint32_t TtlSizeCalc_FOOBAR(FOOBAR* record) {
  return (4 + record->n);
}

